rfcs/037-index-query-composition.md

```diff
@@ -0,0 +1,115 @@
+4. **Where**: this is a component of a query, a single where clause (e.g. `where field_a is not null`). Wheres are connected to columns as well as to queries. By connecting to queries, Amundsen can rank order the most frequent wheres by accessing the query executions.
+
+## UI/UX-level Explanation
+
+Amundsen will display the most common filters and frequent joins based off of query execution counts:
+
+Common filters:
+
+![Common Filters](../assets/037/common-filters.png)
+
+Display common joins:
+
+![Common Joins](../assets/037/common-joins.png)
+
+Queries can also be displayed for tables similar to how queries are shown for dashboards.
+
+## Reference-level Explanation (aka Technical Details)
+
+Four new models will be added to support the data assets listed above:
+
+1. Query
+    - Attributes:
+        - sql: the SQL code for the query
+        - tables: a list of Table Metadata representing each table used in the query
+        - clean_sql: an optional SQL string that the implementer has curated. For example, if an Airflow DAG injects the date into a SQL query every day, the implementer may strip that date and pass the updated SQL into `clean_sql` in order to allow Amundsen to identify that this is not a unique query.
+        - user: an optional user that executed the query (should be Amundsen user ID)
+    - Relations:
+        - Table > Query
+        - User > Query
+2. Query Execution
+    - Attributes:
+        - query: the query model defined above
+        - start time: The start time of the window that the aggregation represents. Query executions can be dropped from the database when start time > X in order to update the most commonly used joins and wheres.
+        - execution_count: the total # of times the query was executed in the window
+        - window_duration: a string representation of the window length. Amundsen will not enforce non-overlapping windows, users should schedule a job to create query executions on defined cadence.
+    - Relations:
+        - Query >> Query Execution
+3. Join
+    - Attributes:
+        - left_table: Table Metadata for the left table in a join
+        - right_table: Table Metadata for the right table in a join
+        - left_column: Column Metadata for the left column in a join
+        - right_column: Column Metadata for the right column in a join
+        - join_type: a string representing the join type, could be `inner join`, `left outer join`, etc
+        - join_operator: a string representing the operator, could be `=`, `>`, `<=`, etc.
+        - join_sql: the sql code for this join (e.g. `table join table2 on table.a = table2.b`)
+        - query: The query model defined above
+    - Relations:
+        - Column >> Join
+        - Query >> Join
+4. Where
```

Comment:
again, I am not sure about whether this is useful. Could you talk about some real-life example from an user standpoint how user to use this where clause info?

Reply:
added in other thread https://github.com/amundsen-io/rfcs/pull/37#discussion_r632748662